### PR TITLE
[@types/epilogue] Create type definition for Epilogue

### PR DESCRIPTION
**Pull request by micksatana**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/epilogue/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+export class EpilogueError extends Error {
+    name: string;
+    message: string;
+    errors: string[];
+    status: EpilogueError | number;
+    cause: Error;
+}
+
+export function initialize(options: InitializeOptions): undefined;
+export function resource(options: ResourceOptions): Resource;
```
Comment (plantain-00) [inline types/epilogue/index.d.ts]: The `options` seems to be optional.
https://github.com/dchester/epilogue/blob/master/lib/index.js#L14
https://github.com/dchester/epilogue/blob/master/lib/index.js#L40

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Review by plantain-00 — APPROVED** [review]

**Comment by micksatana** [comment]
Thanks @plantain-00 

Updated.

File: types/epilogue/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    search?: ResourceSearchOption;
+    sort?: ResourceSortOption;
+    reloadInstances?: boolean;
+    associations?: AssociationOptions;
+    excludeAttributes?: string[];
+    readOnlyAttributes?: string[];
+    updateMethod?: string;
+}
+
+export function initialize(options?: InitializeOptions): undefined;
```
Comment (uniqueiniquity) [inline types/epilogue/index.d.ts]: Should the return type here be `void`?

Comment (micksatana) [reply types/epilogue/index.d.ts]: Thanks @uniqueiniquity 

Done.

**Pull request merged**